libmscore/album.cpp

```diff
@@ -263,14 +396,108 @@ bool Album::saveToFile(const QString &path)
 //   writeAlbum
 //---------------------------------------------------------
 
-void Album::writeAlbum(XmlWriter &writer)
+void Album::writeAlbum(XmlWriter &writer, bool absolutePathEnabled)
 {
     writer.stag("Album");
-    writer.tag("name", _albumTitle);
-    for (auto& aItem : _albumItems) {
-        aItem->writeAlbumItem(writer);
+    writer.tag("name", m_albumTitle);
+    for (auto& aItem : m_albumItems) {
+        aItem->writeAlbumItem(writer, absolutePathEnabled);
     }
     writer.etag();
 }
 
+//---------------------------------------------------------
+//   albumItems
+//---------------------------------------------------------
+
+std::vector<AlbumItem*> Album::albumItems() const
```

Comment:
I'm interested, what use case do you have for a copy of `m_albumItems`?

Reply:
Originally opening a score would add it to `MuseScoreCore::scoreList`. This also opened a new tab for that Score. This is obviously undesired here because if an Album contains 100 Score we don't want 101 tabs for album mode. We also don't want closing a tab to break our Album (because the AlbumItem would suddenly have no score). The solution to this is to have 2 different `openScore` functions one for adding the score to the `scoreList` and creating a new tab and one for creating a new `AlbumItem`. `albumItems()` is used to check whether score that is not in the `scoreList` is open in an `albumItem`.

Reply:
Ok, but why can't it just return a reference to `m_albumItems`? Looking through the full code, I don't see you storing the value returned by this function anywhere, but I may have missed something.

Reply:
@SKefalidis ping :)

Reply:
Hm... I can't remember why I did this. Might have been to not mess with smart pointers outside the scope of this class (since they are not used in many places having the Album return smart pointers looks a bit weird).